lib/widgets/home_body_content.dart

```diff
@@ -42,21 +44,50 @@ class HomeBodyContent extends StatelessWidget {
     );
   }
 
-  List<Widget> buildList(AsyncSnapshot snapshot) {
+  List<Widget> buildList(BuildContext context, AsyncSnapshot snapshot) {
     return List.generate(snapshot.data.results.length, (index) {
-      return Card(
-        color: Colors.black,
-        shadowColor: Colors.white,
-        elevation: MeasuresConstants.cardElevation,
-        child: FadeInImage.assetNetwork(
-          placeholder: StringConstants.defaultPoster,
-          image: snapshot.data.results[index].posterPath != null
-              ? StringConstants.uriPosterImage +
-                  snapshot.data.results[index].posterPath
-              : StringConstants.defaultPoster,
-          fit: BoxFit.cover,
+      return InkWell(
+        onTap: () {
+          Navigator.push(
+            context,
+            MaterialPageRoute(
+              builder: (
+                context,
+              ) =>
+                  MovieDetailPage(
+                result: snapshot.data.results[index],
+                functionGetImage: getImage,
+              ),
+            ),
+          );
+        },
+        child: Card(
+          color: Colors.black,
+          shadowColor: Colors.white,
+          elevation: MeasuresConstants.cardElevation,
+          child: Hero(
+            tag: StringConstants.heroMovieDetailTransitionTag + snapshot.data.results[index].id.toString(),
+            child: getImage(
+              posterPath: snapshot.data.results[index].posterPath,
+              id: snapshot.data.results[index].id.toString(),
+            ),
+          ),
         ),
       );
     });
   }
+
+  FadeInImage getImage({
+    String posterPath,
+    String defaultPath = StringConstants.defaultPoster,
+    String id,
+  }) {
+    return FadeInImage.assetNetwork(
+      placeholder: defaultPath,
+      image: posterPath != null
+          ? StringConstants.uriPosterImage + posterPath
+          : defaultPath,
+      fit: BoxFit.cover,
+    );
+  }
```

Comment:
You are using in two different places, here in home_body_content and in movie_detail_image and basically is the same code so you can create a new widget for this.

Reply:
Done.